### PR TITLE
ready to be reviewed

### PR DESCRIPTION
**Pull request by d-r-holden**

### CODE REVIEW

File: password_validator.rb [hunk 1]
```diff
@@ -3,5 +3,17 @@
...
+    return true
+  elsif password.length >= 8 && password.include?('$')
+    return true
+  elsif password.length >= 8 && password.include?('%')
+    return true
+  elsif password.length >= 8 && password.include?('&')
+    return true
+  else
+    return false
+end
```
Comment (S-McD) [inline password_validator.rb:18]: Don't forget to indent this penultimate 'end' so it won't get deleted by mistake

File: password_validator.rb [hunk 2]
```diff
@@ -3,5 +3,17 @@
...
-  
+  if password.length >= 8 && password.include?('!')
+    return true
+  elsif password.length >= 8 && password.include?('@')
+    return true
+  elsif password.length >= 8 && password.include?('$')
+    return true
+  elsif password.length >= 8 && password.include?('%')
+    return true
+  elsif password.length >= 8 && password.include?('&')
```
Comment (S-McD) [inline password_validator.rb:14]: These if statements are great but there are ways to refactor them. One example is the method I used and the other is shown on the refactoring page which is next in the tutorials. They'll save loads of time once you get the hang of them

**Review by S-McD — COMMENTED** [review]
Hi Daniel, please see my review and well done on completing the challenge

File: password_validator.rb (hunk 2 above)
Comment (d-r-holden) [reply password_validator.rb:14]: Thanks - its definitely something I'm going to read up on and look to implement more.

File: password_validator.rb (hunk 1 above)
Comment (d-r-holden) [reply password_validator.rb:18]: Very good point - thank you

File: password_validator.rb (hunk 2 above)
Comment (S-McD) [reply password_validator.rb:14]: Perhaps the following resource will help:
https://www.rubyguides.com/2015/06/ruby-regex/

Comment (d-r-holden) [reply password_validator.rb:14]: brilliant - thank you

**Pull request merged**